content/blog/2019/07/2019-07-30-jenkins-cli.adoc

```diff
@@ -0,0 +1,88 @@
+---
+layout: post
+title: "Managing Jenkins with jcli"
+tags:
+- general
+- jcli
+author: linuxsuren
+---
+
+As a developer, I usually use Jenkins like this:
+
+* Find a job which is related with your current work
+* Trigger that job
+* Check the output of log
+
+Sometimes, I might need to check the update center. Maybe a new plugin is needed, or update an exists plugin.
+Or, I want to upload a plugin from my computer. For all these cases, I just don't need a UI or even a browse.
+I like to use a CLI to complete most of my tasks. For example, I use `kubectl` to manage my Kubernetes cluster,
+to create or modify the kubernetes resources. So, I start to think that why not using a CLI to manage my Jenkins.
+
+== Why create a new one?
+
+First, I almost forget the Jenkins CLI is existed which is written by Java. Let me do dome introduce how to use the exist one.
+
+Visit Jenkins page from `http://localhost:8080/jenkins/cli/`. You'll see a command like `java -jar jenkins-cli.jar -s http://localhost:8080/jenkins/ help`. So, a jar file needs to be download.
+We can use this command to complete this task `wget http://localhost:8080/jenkins/jnlpJars/jenkins-cli.jar`.
+
+Now, I believe you can see this's not a linux cli style. Please consider some points below:
+
+* The users must have a JRE. It's not convenience for other developers who don't use Java.
+* Type more words. We always need to type `java -jar jenkins-cli.jar -s http://localhost:8080/jenkins/` as the inital cmd.
+* Cannot install it by some popular package manager, like `brew` or `yum`.
+
+Of course, the Java CLI client is more native with Jenkins. But I'd like to use this more easier.
+
+That's the story of creating `jcli`.
+
+== Features
+
+* Easy to maintain config file for jcli
+* Multiple Jenkins support
+* Plugins management (list, search, install, upload)
+* Job management (search, build, log)
+* Open your Jenkins with a browse
+* Restart your Jenkins
+* Connection with proxy support
+
+== How to get it?
+
+You can find `jcli` from this repo https://github.com/jenkins-zh/jenkins-cli/[jenkins-cli]. For now, we support 
+these three most popular os platform.
+
+=== MacOS
+
+You can use brew to install jcli.
+
+```
+brew tap jenkins-zh/jcli
+brew install jcli
+```
+
+=== Linux
+
+It's very simple to install jcli into your Linux OS. Just need to execute a command line at below:
+
+```
+curl -L https://github.com/jenkins-zh/jenkins-cli/releases/latest/download/jcli-linux-amd64.tar.gz|tar xzv
+sudo mv jcli /usr/local/bin/
+```
+
+=== Windows
+
+You can find the latest version by click https://github.com/jenkins-zh/jenkins-cli/releases/latest/download/jcli-windows-386.tar.gz[here]. Then download the tar file, cp the uncompressed `jcli` directory into your system path.
+
+=== How to get started?
+
+It's very simple to use this. Once you got `jcli` in your computer. One thing is needed, use below cmd to generate and 
```

Comment:
"Once you get jcli on your computer, use this command to generate and modify your Jenkins configuration:"